polkadot/node/core/pvf/prepare-worker/src/lib.rs

```diff
@@ -306,21 +329,96 @@ struct JobResponse {
 	memory_stats: MemoryStats,
 }
 
+#[cfg(target_os = "linux")]
+fn handle_clone(
+	pvf: &PvfPrepData,
+	pipe_write_fd: i32,
+	pipe_read_fd: i32,
+	stream_fd: i32,
+	preparation_timeout: Duration,
+	prepare_job_kind: PrepareJobKind,
+	executor_params: &Arc<ExecutorParams>,
+	worker_info: &WorkerInfo,
+	temp_artifact_dest: &Path,
+	usage_before: Usage,
+) -> Result<PrepareWorkerSuccess, PrepareError> {
+	use polkadot_node_core_pvf_common::worker::security;
+
+	// 2MiB * num_threads
+	let stack_size = 2 * 1024 * 1024 * PREPARE_WORKER_THREAD_NUMBER;
```

Comment:
Stack size is a per-thread entity by definition. Threads cannot share stack. Thus, process stack size (to be precise, the stack size of the process's main thread) cannot depend on the number of threads. Unless I miss something, some logical mistake should be corrected here.

Reply:
You are right, I made a [little experiment](https://github.com/mrcnski/clone-stack-experiment) to correct my understanding. So for the clone stack size we can always just use 2 MiB, or is that even overkill?

Reply:
Indeed, it is a shame that `nix` strictly requires a pre-allocated stack here. That's because it uses libc's `clone()` wrapper instead of raw `clone3()` syscall. You wouldn't need to specify explicit stack at all as you're not sharing virtual address space with the cloned process; the stack of the parent process would just be cloned as well as all the rest.

Still, we have what we have, and the child process requires some reasonable stack. Offhand, 2 Mb looks okay to me.